### DISCUSSION TITLE
Installing > Hash Sum mismatch

### DISCUSSION THREAD

**Discussion by choozuradventure** [abraunegg/onedrive#2572] [Q&A] [answered]
I'm a Linux novice trying to install this application on a Raspberry Pi.  Going through the installation instructions it points me to [Installation of 'onedrive' package on Debian and Ubuntu](https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md)

Below is what happens when I try to install.  Any ideas where I went wrong?

P.S. a couple of earlier steps in the installation instructions also don't seem to work on Raspberry Pi.

```
sudo add-apt-repository --remove ppa:yann1ck/onedrive
sudo: add-apt-repository: command not found
```
and
```
Created symlink /etc/systemd/user/default.target.wants/onedrive.service ? /usr/lib/systemd/user/onedrive.service.
bash: Created: command not found
```

Hash Sum mismatch:

```
sudo apt install --no-install-recommends --no-install-suggests onedrive
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following additional packages will be installed:
  libphobos2-ldc-shared100
The following NEW packages will be installed:
  libphobos2-ldc-shared100 onedrive
0 upgraded, 2 newly installed, 0 to remove and 0 not upgraded.
Need to get 614 kB/1,759 kB of archives.
After this operation, 8,564 kB of additional disk space will be used.
Do you want to continue? [Y/n] Y
Get:1 https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/Debian_12 ./ onedrive 2.4.25-1+np1 [614 kB]
Err:1 https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/Debian_12 ./ onedrive 2.4.25-1+np1
  Hash Sum mismatch
  Hashes of expected file:
   - SHA256:10645ee96612e67fe6bd73fd8525969ec50fb39f77b39c04b5f5bc158d3f1ba7
   - SHA1:dbce109318c7f44ed26c9b894ebdc105b9fbe473 [weak]
   - MD5Sum:cbe9af766fc4d5403220b65cf7121189 [weak]
   - Filesize:613896 [weak]
  Hashes of received file:
   - SHA256:f17b1460e384d4259422919079e11a4080ac9e73e7215a734596982716e35a43
   - SHA1:4013e0c71c83be921f933de3f2f9e455f64f91f8 [weak]
   - MD5Sum:1961463eaa19960f6d7f021fa34f2bde [weak]
   - Filesize:613896 [weak]
  Last modification reported: Sat, 07 Oct 2023 12:35:54 +0000
Fetched 614 kB in 2s (341 kB/s)   
E: Failed to fetch https://provo-mirror.opensuse.org/repositories/home%3A/npreining%3A/debian-ubuntu-onedrive/Debian_12/arm64/onedrive_2.4.25-1%2Bnp1_arm64.deb  Hash Sum mismatch
   Hashes of expected file:
    - SHA256:10645ee96612e67fe6bd73fd8525969ec50fb39f77b39c04b5f5bc158d3f1ba7
    - SHA1:dbce109318c7f44ed26c9b894ebdc105b9fbe473 [weak]
    - MD5Sum:cbe9af766fc4d5403220b65cf7121189 [weak]
    - Filesize:613896 [weak]
   Hashes of received file:
    - SHA256:f17b1460e384d4259422919079e11a4080ac9e73e7215a734596982716e35a43
    - SHA1:4013e0c71c83be921f933de3f2f9e455f64f91f8 [weak]
    - MD5Sum:1961463eaa19960f6d7f021fa34f2bde [weak]
    - Filesize:613896 [weak]
   Last modification reported: Sat, 07 Oct 2023 12:35:54 +0000
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
```

**Comment by abraunegg** [comment, marked as answer]
@choozuradventure 
This is a Debian package issue. Please raise your concern on the Debian Support Forums.

I have zero control over any packages.

**Reply by choozuradventure** [reply]
Ok will do.  Is [Debian User Forums](https://forums.debian.net/) the right place?

Also just to be be sure, I was trying to install the correct package for a Respberry Pi, correct?

**Reply by abraunegg** [reply]
Yes, you were trying to install the correct package.

The issue is with the mirror - so, you need to seek support on how to use a different mirror.

**Comment by russfink** [comment]
Having the same "hash sum mismatch" problem on Debian 12 Bookworm as of 4-Jan-2024 13:13 GMT, as well as yesterday.  Fixed by bypassing the apt installation and downloading the .deb package directly:

    wget https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/Debian_12/amd64/onedrive_2.4.25-1+np1_amd64.deb
    sudo apt install libphobos2-ldc-shared100
    sudo dpkg -i onedrive_2.4.25-1+np1_amd64.deb 

I'm guessing this is not a Debian issue but an OpenSuse repository issue since everything else works fine except for the package itself.  I'm trusting that the problem is that the package file date is showing as 2023-12-10 per the download link but that the hashes in the repository were last updated on 2023-10-07 per apt output.
